functions.php

```diff
@@ -0,0 +1,98 @@
+<?php
+require_once 'dbConn.php';
+
+
+/*
+ * Is used to select data from the database and return it as an array
+ *
+ * @param1 - PDO - this is where the $db var should be used otherwise the function can't access the database(scope)
+ * @param2 - $table - used to choose which table you want from the db, should be used in double quotes with backticks.
+ * @param3 - $field - which field you want from the table, in double quotes with backticks
+ *
+ * @return - an array - should be an array of the field that's picked.
+ */
+function retrieveData ($db, $table, $field) :array
+{
+    $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+    $query = $db->prepare("SELECT $field FROM $table;");
+    $query->execute();
+    $result = $query->fetchAll();
+    if (is_array($result) == true) {
+        return $result;
+    }
+}
+
+///*
+// * Is used to turn the array result into text that can be displayed on a page
+// *
+// * @param1 - array - this should be the array that you want to be text
+// * @param2 - array key - this is the key of the array item you want to be displayed as text, as it will be found in the db
+// *
+// * @return - string - this will be the string pulled out of the array item
+// */
+//function arrayToText ($array, $key) :string
+//{
+//    if (array_key_exists($key, $array) == true && is_string($key) == true) {
+//        return "$array[$key]";
+//    } else {
+//        return 'failed to turn into text';
+//    }
+//}
+
+/*
+ * This function is used to add a new project to the db
+ *
+ * @param1 - PDO - this is where the database variable will be inputted into the array
```

Comment:
take a look at how to doc block, need to identify which param by name

Reply:
done